logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+#define LOGS_TABLE "Logs"
+#define BLOBS_TABLE "Blobs"
+
+static uv_loop_t *db_loop;
+static sqlite3 *main_db;
+static char *main_db_dir;  /**< Directory where all the log management databases and log blobs are stored in **/
+static char *main_db_path; /**< Path of MAIN_DB **/
+
+void db_set_lock(uv_mutex_t *db_mut) { 
+	uv_mutex_lock(db_mut); 
+}
+
+void db_release_lock(uv_mutex_t *db_mut) { 
+	uv_mutex_unlock(db_mut);
+}
+
+/**
+ * @brief Throws fatal SQLite3 error
+ * @details In case of a fatal SQLite3 error, the SQLite3 error code will be 
+ * translated to a readable error message and logged to stderr. 
+ * @param[in] rc SQLite3 error code
+ * @param[in] line_no Line number where the error occurred 
+ */
+static inline void fatal_sqlite3_err(int rc, int line_no){
+	debug(D_LOGS_MANAG, "SQL ERROR:%s", sqlite3_errstr(rc));
+	fatal("SQLite error: %s (line %d)", sqlite3_errstr(rc), line_no);
+}
+
+/**
+ * @brief Throws fatal libuv error
+ * @details In case of a fatal libuv error, the libuv error code will be 
+ * translated to a readable error message and logged to stderr. 
+ * @param[in] rc libuv error code
+ * @param[in] line_no Line number where the error occurred 
+ */
+static inline void fatal_libuv_err(int rc, int line_no){
+	fatal("libuv error: %s (line %d)", uv_strerror(rc), line_no);
+}
+
+/**
+ * @brief Get version of SQLite
+ * @return String that contains the SQLite version. Must be freed.
+ */
+char *db_get_sqlite_version() {
+    int rc = 0;
+    sqlite3_stmt *stmt_get_sqlite_version;
+    rc = sqlite3_prepare_v2(main_db,
+                            "SELECT sqlite_version();", -1, &stmt_get_sqlite_version, NULL);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+    rc = sqlite3_step(stmt_get_sqlite_version);
+    if (unlikely(rc != SQLITE_ROW)) fatal_sqlite3_err(rc, __LINE__);
+    char *text = mallocz(sqlite3_column_bytes(stmt_get_sqlite_version, 0) + 1);
+    strcpy(text, (char *)sqlite3_column_text(stmt_get_sqlite_version, 0));
+    rc = sqlite3_finalize(stmt_get_sqlite_version);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+    return text;
+}
+
+static void db_writer(void *arg){
+	int rc = 0;
+	struct File_info *p_file_info = *((struct File_info **) arg);
+	
+	uv_loop_t *writer_loop = mallocz(sizeof(uv_loop_t));
+    rc = uv_loop_init(writer_loop);
+    if (unlikely(rc)) fatal_libuv_err(rc, __LINE__);
+
+	/* Prepare LOGS_TABLE INSERT statement */
+	sqlite3_stmt *stmt_logs_insert;
+	rc = sqlite3_prepare_v2(p_file_info->db,
+                        "INSERT INTO " LOGS_TABLE "("
+                        "FK_BLOB_Id,"
+                        "BLOB_Offset,"
+                        "Timestamp,"
+                        "Msg_compr_size,"
+                        "Msg_decompr_size"
+                        ") VALUES (?,?,?,?,?) ;",
+                        -1, &stmt_logs_insert, NULL);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
```

Comment:
We are having a different approach to deal with prepared statements in our database, please, take a look in [this](https://github.com/netdata/netdata/blob/master/database/sqlite/sqlite_context.c) file and try to use the same for we have only one pattern in our code.